lib/models/SearchCriteria.dart

```diff
@@ -0,0 +1,47 @@
+import 'package:intl/intl.dart';
+
+class SearchCriteria {
+  String keyword = "";
+  DateTime? startDate;
+  DateTime? endDate;
+
+  // Constructor
+  SearchCriteria(this.keyword, this.startDate, this.endDate);
+
+  // Constructor to parse params
+  SearchCriteria.withList(List<String> params) {
+    DateTime? _potentialStart;
+    DateTime? _potentialEnd;
+    String _potentialKeyword = "";
+
+    if (params.isEmpty) {
+      // default init
+      _potentialStart = DateTime.now();
+      _potentialEnd = DateTime.now();
```

Comment:
There is a null check, and fallback to `DateTime.now()`, when you assign to `this.startDate` and `this.endDate`.
You can remove this part of the `if` statement and make it `if (params.isNotEmpty) ...` to simplify this.